### DISCUSSION TITLE
Show loader during data fetch

### DISCUSSION THREAD

**Discussion by matobiely** [plouc/nivo#1690] [Q&A]
It there a way, how to show loader, loading dots, or  whatever kind of loader during data fetch?
My problem is, that at the beggining i have empty data `const [data, setData] = useState([]);`
and load them in useEffect. But since fetching take some time, there is no loader and data is empty array, render fails.
I can do loader for whole line chart component, but i want to avoid this kind of loader. I want to show the chart, with axes, custom footer layers etc. during fetching data and after fetching is done, show the lines. 

Thanks

**Comment by wyze** [comment]
You can use a custom layer for that. See an example of custom layer here: https://codesandbox.io/s/fancy-line-chart-3quqe

**Reply by matobiely** [reply]
yeah, but it still doesnt prevent render failature,  because chart still expect data on render, and did not accept empty array. So i though about some property like `loading` | 'loader` and internally it will wait for render/show line until `loading`is `false` but other parts of chart will be shown
 
```
const [loading, setLoading] = useState(true);
const [data, setData] = useState([]);

const loadData = () => {
  //...fetchData
setData(res)
setLoading(false)
}

<Line
...
data={data}
loading={loading}
/>
```

**Comment by NewbiZ** [comment]
That would be great indeed. I'm having the same need and conditionally replacing the whole graph with a spinner makes layout changes.

**Comment by tkonopka** [comment]
Hi @matobiely. Is this still an open question? Seems to me the problematic thing here would be if an empty dataset caused a rendering error. An error would indeed prevent using a custom layer. However, the sandbox provided by @wyze works fine with `data={[]}`. So it should be possible to draw a custom layer to display a loading message, and remove that message when the data becomes non-empty. Am I missing something? Or is the chart failure you mentioned occurring with another chart type?
